ui/src/app/job-details/tabs/timing-diagram/timing-diagram.component.ts

```diff
@@ -30,36 +30,28 @@ export class JobTimingDiagramComponent implements OnInit {
     let chartRows = 1;
 
     metadata.forEach((task) => {
-      if (task.shards) {
-        task.shards.forEach((shard) => {
-          if (shard.executionEvents) {
-            shard.executionEvents.forEach((event) => {
-              this.timelineChart.dataTable.push(this.formatRow(task.name + ' shard ' + shard.shardIndex, event.name, event));
-            });
-          } else {
-            this.timelineChart.dataTable.push(this.formatRow(task.name + ' shard ' + shard.shardIndex, task.name, shard));
-          }
-          chartRows++;
-        });
-      } else if (task.name && task.start) {
-        if (task.executionEvents) {
+      if (task.shards && task.executionEvents) {
+        task.executionEvents.forEach((event) => {
+              this.timelineChart.dataTable.push(this.formatRow(task.name + ' shard ' + event.shardIndex + ' attempt ' + event.attemptNumber, event.name, event));
+          });
+        chartRows = chartRows + task.shards.length;
+      } else if (task.name && task.start && task.executionEvents) {
           task.executionEvents.forEach((event) => {
-            this.timelineChart.dataTable.push(this.formatRow(task.name, event.name, event));
+            this.timelineChart.dataTable.push(this.formatRow(task.name + ' attempt ' + event.attemptNumber, event.name, event));
           });
-        } else {
-          this.timelineChart.dataTable.push(this.formatRow(task.name, task.name, task));
-        }
         chartRows++;
       }
     });
 
+    console.log('rows: ' + chartRows);
+
```

Comment:
This looks like a lingering debug statement?

Reply:
it was – removed